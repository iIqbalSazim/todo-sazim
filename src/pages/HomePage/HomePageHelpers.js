export function findIndexWithId(data, id) {
  return data.findIndex((element) => element.id === id);
}

export function formatCompletedAt(dateString) {
  const date = new Date(dateString);

  let currentTime =
    date.toLocaleTimeString().slice(0, 4) + date.toLocaleTimeString().slice(7);
  let currentDate = `${currentTime} | ${date.toDateString()}`;
  return currentDate;
}

export function formatDueDate(dateString) {
  const date = new Date(dateString);
  return date.toDateString().slice(0, 15);
}

export const replaceTabId = (htmlString) => {
  return htmlString.replace(
    /id="mantine-[a-z0-9]+-tab-(all|active|completed|target)"/g,
    (match, p1) => `id="mantine-mockedId-tab-${p1}"`
  );
};

export const replaceMenuTargetId = (htmlString) => {
  return htmlString.replace(
    /id="mantine-[a-z0-9]+-(target)"/g,
    (match, p1) => `id="mantine-mockedId-${p1}"`
  );
};

export const replaceDropdownTargetId = (htmlString) => {
  return htmlString.replace(
    /id="mantine-[a-z0-9]+-(dropdown)"/g,
    (match, p1) => `id="mantine-mockedId-${p1}"`
  );
};

export const replaceDataFocusId = (htmlString) => {
  return htmlString.replace(
    /data-focus-id="mantine-[a-z0-9]+"/g,
    'data-focus-id="mantine-mockedId"'
  );
};

export const replaceAriaLabelledBy = (htmlString) => {
  return htmlString.replace(
    /aria-labelledby="mantine-[a-z0-9]+-target"/g,
    'aria-labelledby="mantine-mockedId-target"'
  );
};

export const replaceAriaControls = (htmlString) => {
  return htmlString.replace(
    /aria-controls="mantine-[a-z0-9]+-dropdown"/g,
    'aria-controls="mantine-mockedId-dropdown"'
  );
};

export const replaceAllRandomGeneratedIds = (htmlString) => {
  let result = replaceTabId(htmlString);
  result = replaceMenuTargetId(result);
  result = replaceDropdownTargetId(result);
  result = replaceDataFocusId(result);
  result = replaceAriaLabelledBy(result);
  result = replaceAriaControls(result);

  return result;
};
