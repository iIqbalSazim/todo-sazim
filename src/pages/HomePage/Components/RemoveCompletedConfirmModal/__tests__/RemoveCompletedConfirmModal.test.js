import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { replaceAllRandomGeneratedIds } from "../../../HomePageHelpers";
import RemoveCompletedConfirmModal from "../RemoveCompletedConfirmModal";

const mockClearAllCompletedTasks = jest.fn();
const mockCloseModal = jest.fn();

describe("RemoveCompletedConfirmModal", () => {
  test("renders correctly", () => {
    const { container } = render(
      <RemoveCompletedConfirmModal
        clearAllCompletedTasks={mockClearAllCompletedTasks}
        closeModal={mockCloseModal}
        isConfirmModalOpen
      />
    );

    const titleElement = screen.getByRole("heading", {
      name: /confirm delete/i,
    });
    const areYouSureElement = screen.getByText(
      "Are you sure you want to permanently delete all completed tasks?"
    );
    const keepThemButton = screen.getByRole("button", {
      name: "No, Keep them.",
    });
    const deleteButton = screen.getByRole("button", {
      name: "Yes, Delete!",
    });

    expect(titleElement).toBeInTheDocument();
    expect(areYouSureElement).toBeInTheDocument();
    expect(keepThemButton).toBeInTheDocument();
    expect(deleteButton).toBeInTheDocument();

    expect(replaceAllRandomGeneratedIds(container.innerHTML)).toMatchSnapshot();
  });

  test("calls closeModal function on 'No, Keep them.' button click", async () => {
    render(
      <RemoveCompletedConfirmModal
        clearAllCompletedTasks={mockClearAllCompletedTasks}
        closeModal={mockCloseModal}
        isConfirmModalOpen
      />
    );

    const keepThemButton = screen.getByRole("button", {
      name: "No, Keep them.",
    });

    await userEvent.click(keepThemButton);

    expect(mockCloseModal).toHaveBeenCalled();
  });

  test("calls clearAllCompletedTasks and closeModal functions on 'Yes, Delete!' button click", async () => {
    render(
      <RemoveCompletedConfirmModal
        clearAllCompletedTasks={mockClearAllCompletedTasks}
        closeModal={mockCloseModal}
        isConfirmModalOpen
      />
    );

    const deleteButton = screen.getByRole("button", {
      name: "Yes, Delete!",
    });

    await userEvent.click(deleteButton);

    expect(mockClearAllCompletedTasks).toHaveBeenCalled();
    expect(mockCloseModal).toHaveBeenCalled();
  });
});
