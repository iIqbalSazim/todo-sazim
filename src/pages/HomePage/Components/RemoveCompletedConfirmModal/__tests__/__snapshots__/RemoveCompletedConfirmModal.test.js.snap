// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveCompletedConfirmModal renders correctly 1`] = `""`;
